corim/signedcorim.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2021 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package corim
+
+import (
+	"crypto"
+	"crypto/rand"
+	"errors"
+	"fmt"
+
+	cose "github.com/veraison/go-cose"
+)
+
+var (
+	ContentType = "application/rim+cbor"
+)
+
+// SignedCorim encodes a signed-corim type (i.e., a COSE Sign1 wrapped CoRIM)
+// with signature and verification methods
+type SignedCorim struct {
+	UnsignedCorim UnsignedCorim
+	message       *cose.Sign1Message
+}
+
+func (o *SignedCorim) processHdrs() error {
+	var hdr = o.message.Headers
+
+	if hdr.Protected == nil {
+		return errors.New("missing mandatory protected header")
+	}
+
+	v, ok := hdr.Protected[3]
```

Comment:
Is the 3 here obvious? Or perhaps for the new comers to COSE, we need a better description of the same as a suitable constant!

Reply:
good point!  See affaf2b

Reply:
Thank You! Looks very good now!